sw/vendor/meson.build

```diff
@@ -9,3 +9,59 @@ vendor_coremark_base_files = files([
   'eembc_coremark/core_state.c',
   'eembc_coremark/core_util.c',
 ])
+
+# Googletest and Googlemock Build Instructions
+thread_dep = dependency('threads')
+sw_vendor_gtest_dir = 'google_googletest'
+
+# Begin Based on https://github.com/mesonbuild/gtest
+sw_vendor_gtest_googletest_dir = sw_vendor_gtest_dir / 'googletest'
+
+sw_vendor_gtest_googletest_inc_dir = include_directories(
+  sw_vendor_gtest_googletest_dir / 'include',
+  sw_vendor_gtest_googletest_dir,
+  is_system: true,
+)
+sw_vendor_gtest_googletest_libsources = files(sw_vendor_gtest_googletest_dir / 'src/gtest-all.cc')
+sw_vendor_gtest_googletest_mainsources = files(sw_vendor_gtest_googletest_dir / 'src/gtest_main.cc')
+
+sw_vendor_gtest_googletest_dep = declare_dependency(
+  include_directories : sw_vendor_gtest_googletest_inc_dir,
+  sources: sw_vendor_gtest_googletest_libsources,
+  dependencies: thread_dep,
+)
+
+sw_vendor_gtest_googletest_main_dep = declare_dependency(
+  include_directories: sw_vendor_gtest_googletest_inc_dir,
+  sources: [
+    sw_vendor_gtest_googletest_libsources,
+    sw_vendor_gtest_googletest_mainsources,
+  ],
+  dependencies: thread_dep,
+)
+
+sw_vendor_gtest_googlemock_dir = sw_vendor_gtest_dir / 'googlemock'
+
+sw_vendor_gtest_googlemock_inc_dir = include_directories(
+  sw_vendor_gtest_googlemock_dir / 'include',
+  sw_vendor_gtest_googlemock_dir,
+  is_system: true
+)
+sw_vendor_gtest_googlemock_libsources = files(sw_vendor_gtest_googlemock_dir / 'src/gmock-all.cc')
+sw_vendor_gtest_googlemock_mainsources = files(sw_vendor_gtest_googlemock_dir / 'src/gmock_main.cc')
+
+sw_vendor_gtest_googlemock_dep = declare_dependency(
+  include_directories: sw_vendor_gtest_googlemock_inc_dir,
+  sources: sw_vendor_gtest_googlemock_libsources,
+  dependencies: thread_dep,
+)
+
+sw_vendor_gtest_googlemock_main_dep = declare_dependency(
+  include_directories: sw_vendor_gtest_googlemock_inc_dir,
+  sources: [
+    sw_vendor_gtest_googlemock_libsources,
+    sw_vendor_gtest_googlemock_mainsources,
+  ],
+  dependencies: thread_dep,
+)
```

Comment:
This part can be simplified (please see below) to declare a single dependency with googletest lib sources, googlemock lib sources, and googlemock main.cc. Then, we can remove `sw_vendor_gtest` in `sw/device/lib/testing/meson.build` and depend directly to `sw_vendor_gtest`.
```

sw_vendor_gtest_googlemock_dir = sw_vendor_gtest_dir / 'googlemock'

sw_vendor_gtest_googlemock_inc_dir = include_directories(
  sw_vendor_gtest_googlemock_dir / 'include',
  sw_vendor_gtest_googlemock_dir,
  is_system: true
)
sw_vendor_gtest_googlemock_libsources = files(sw_vendor_gtest_googlemock_dir / 'src/gmock-all.cc')
sw_vendor_gtest_googlemock_mainsources = files(sw_vendor_gtest_googlemock_dir / 'src/gmock_main.cc')

sw_vendor_gtest = declare_dependency(
  include_directories: [
    sw_vendor_gtest_googletest_inc_dir,
    sw_vendor_gtest_googlemock_inc_dir,
  ],  
  sources: [
    sw_vendor_gtest_googletest_libsources,
    sw_vendor_gtest_googlemock_libsources,
    sw_vendor_gtest_googlemock_mainsources,
  ],  
  dependencies: thread_dep,
)
```

Reply:
I wasn't sure if other build targets would want googletest (which would be a reason to keep it so simple), but I'm happy to simplify these definitions along the lines of your suggestion, given it won't be so hard to add other definitions later.

Reply:
Done